src/rgw/rgw_sync.cc

```diff
@@ -1459,17 +1459,19 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
 
       if (!lost_lock) {
         /* update marker to reflect we're done with full sync */
-        if (can_adjust_marker) yield {
-          sync_marker.state = rgw_meta_sync_marker::IncrementalSync;
-          sync_marker.marker = sync_marker.next_step_marker;
-          sync_marker.next_step_marker.clear();
-
-          RGWRados *store = sync_env->store;
-          ldout(sync_env->cct, 0) << *this << ": saving marker pos=" << sync_marker.marker << dendl;
-          using WriteMarkerCR = RGWSimpleRadosWriteCR<rgw_meta_sync_marker>;
-          call(new WriteMarkerCR(sync_env->async_rados, store, pool,
-                                 sync_env->shard_obj_name(shard_id),
-                                 sync_marker));
+        if (can_adjust_marker) {
+          yield {
+            sync_marker.state = rgw_meta_sync_marker::IncrementalSync;
+            sync_marker.marker = sync_marker.next_step_marker;
+            sync_marker.next_step_marker.clear();
+
+            RGWRados *store = sync_env->store;
+            ldout(sync_env->cct, 0) << *this << ": saving marker pos=" << sync_marker.marker << dendl;
+            using WriteMarkerCR = RGWSimpleRadosWriteCR<rgw_meta_sync_marker>;
+            call(new WriteMarkerCR(sync_env->async_rados, store, pool,
+                                   sync_env->shard_obj_name(shard_id),
+                                   sync_marker));
+          }
```

Comment:
it looks like the local variable `RGWRados *store` is the only reason this yield {} block is needed. if you take that out, we can just do this:
```c++
            yield call(new WriteMarkerCR(sync_env->async_rados, sync_env->store, pool,
                                         sync_env->shard_obj_name(shard_id),
                                         sync_marker));
```